app/api/resources/admin.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Resources
+  class Admin < Grape::API
+    resource :admin do
+      before do
+        error!('No admin rights', 403) unless current_user.admin?
+      end
+
+      resource :users do
+        desc 'Get current user'
+        get do
+          present User.all, with: Entities::User
+        end
+
+        route_param :id, type: String, desc: 'User id' do
+          desc 'Delete user'
+          delete do
+            delete_self = current_user.id == params[:id]
+            error!('You can not delete current user', 403) if delete_self
+
+            to_delete = User.find(params[:id])
+            error!('User not found', 404) unless to_delete
+            error!(to_delete.errors.messages, 400) unless to_delete.destroy
+
+            status :no_content
+          end
+
+          desc 'Update user'
+          params do
+            requires :data, type: Hash do
+              optional(
+                :role,
+                type: Symbol,
+                default: :user,
+                values: User.roles.symbolize_keys.keys,
```

Comment:
i'm pretty sure db is not queried for accessing all keys (`User.roles`), but since we use here the symbolized keys, im fine with adding this constant :+1: